src/Components/Pages/SetSystemPurposePage.js

```diff
@@ -20,9 +24,16 @@ const SetSystemPurposePage = ({
   isLoading,
   isError,
 }) => {
+  useEffect(() => {
+    if (isEditMode) {
+      setRole(activationKey?.role || '');
+      setSla(activationKey?.serviceLevel || '');
+      setUsage(activationKey?.usage || '');
+    }
+  }, [isEditMode, activationKey, setRole, setSla, setUsage]);
```

Comment:
Suggestion: Try removing the `isEditMode` check. My hunch is that this will work without it (if its not edit mode, the key doesn't exist, so it'll default to the empty string anyway). This coupled with my next suggestion would remove the need to care about edit context in this component